File: Semana_6/Ejercicios/GestionProductos/GestionProducto.java

class GestionProducto {

    class Producto {
        public
        String nombre;
        int precio;
        String categorias;

        Producto(String _nombre, int _precio, String _categorias ) {
            nombre = _nombre;
            precio = _precio;
            categorias = _categorias;
        }

        void mostrar_detalle(){
            System.out.println("Nombre: " + nombre);
        }
    }

    public static void main(String[] args) {
        System.out.println("Hello World");
        
        Producto kevin = new Producto("kevin",19,"Villarica");
        kevin.mostrar_detalle();

    }

}

File: Semana_6/Ejercicios/GestionProductos/GestionProducto.py


class Producto():
    def __init__(self,nombre,precio,categoria):
        self.nombre = nombre
        self.precio = precio
        self.categoria = categoria


    def mostrar_detalle(self):
        print(f"Producto = {self.nombre} Precio = {self.precio} Categoria = {self.categoria}")


class Electronico(Producto):

    def __init__(nombre,precio,categoria,consumo):
        super().__init__(nombre,precio,categoria)
        self.consumo = consumo

    def mostrar_detalle(self):
        print(f"Producto = {self.nombre} Precio = {self.precio} Categoria = {self.categoria} Consumo = {self.consumo}")


class Alimenticio(Producto):

    def __init__(nombre,precio,categoria,calorias):
        super().__init__(nombre,precio,categoria)
        self.calorias = calorias

    def mostrar_detalle(self):
        print(f"Producto = {self.nombre} Precio = {self.precio} Categoria = {self.categoria} Calorias = {self.calorias}")


class Vestimenta(Producto):

    def __init__(nombre,precio,categoria,talla):
        super().__init__(nombre,precio,categoria)
        self.talla = talla

    def mostrar_detalle(self):
        print(f"Producto = {self.nombre} Precio = {self.precio} Categoria = {self.categoria} Talla = {self.talla}")
